### PR TITLE
New API methods including invalidate and custom roots

### PR DESCRIPTION
**Pull request by lynchbomb**
This spike compliments (https://github.com/linkedin/spaniel/pull/80). **We will need to deploy this behind a minor version bump and feature flag.**

This PR includes the following new API method:

- `invalidate`

Additionally:
- Includes the ability to watch a custom root within `onInViewportOnce` and an optional flag `ALLOW_CACHED_SCHEDULER`.
- Bumps Spaniel to v2.5.0

### CODE REVIEW

**Comment by lynchbomb** [comment]
@stefanpenner ^^^^^

**Comment by lynchbomb** [comment]
!Not ready to merge until a new release of Spaniel v2.5.0 is rolled out.

File: addon/services/viewport.js [hunk 1]
```diff
@@ -66,5 +70,29 @@ export default Service.extend({
     if (this._globalWatcher) {
       this._globalWatcher.destroy();
     }
+  },
+
+  invalidate() {
+    spaniel.invalidate();
+  },
+
+  on(eventName, callback, root = window) {
```
Comment (krisselden) [inline addon/services/viewport.js]: what does this API have to do with this service? it doesn't look like it belongs here at all.

File: addon/services/viewport.js [hunk 2]
```diff
@@ -66,5 +70,29 @@ export default Service.extend({
...
+    root.addEventListener(eventName, callback, false);
+  },
+
+  off(eventName, callback, root = window) {
+    root.removeEventListener(eventName, callback, false);
+  },
+
+  onDebounce(eventName, callback, interval, root = window) {
```
Comment (krisselden) [inline addon/services/viewport.js]: there is already a run.debounce util in ember itself. also, this doesn't not use anything in the service and does not seem like it has to do with spaniel.

File: addon/services/viewport.js (hunk 1 above)
Comment (lynchbomb) [reply addon/services/viewport.js]: The intention was solely succinct abstract event binding methods to be called on the viewport. The use-case is to call `spaniel.invalidate()`. Im totally cool with removing `on/off/onDebouce` from the API (this PR) and rather just documenting the implementing leveraging existing API's.

Same answer for:
> there is already a run.debounce util in ember itself. also, this doesn't not use anything in the service and does not seem like it has to do with spaniel.

Comment (krisselden) [reply addon/services/viewport.js]: lets pull on/off/onDebounce outside of this, just document suggested use of other apis. otherwise looks good to me

**Pull request merged**